lib/publish.js

```diff
@@ -36,47 +36,53 @@ module.exports = async (pluginConfig, context) => {
     await Promise.all(
       globbedAssets.map(async asset => {
         const {path, label, type, filepath} = isPlainObject(asset) ? asset : {path: asset};
-        const file = resolve(cwd, path);
-        let fileStat;
-
-        try {
-          fileStat = await stat(file);
-        } catch (_) {
-          logger.error('The asset %s cannot be read, and will be ignored.', path);
-          return;
+        const _url = asset.url
+        if (_url) {
+          assetsList.push({ label, url: _url, type, filepath });
```

Comment:
Unfortunately this will not work as intended.

I wanted to pull these changes into the PR #343 and added an unit test. This is the test I had to write to make it work:
```javascript

test.serial('Publish a release with an asset link', async t => {
  const cwd = 'test/fixtures/files';
  const owner = 'test_user';
  const repo = 'test_repo';
  const env = {GITLAB_TOKEN: 'gitlab_token'};
  const nextRelease = {gitHead: '123', gitTag: 'v1.0.0', notes: 'Test release note body'};
  const options = {repositoryUrl: `https://gitlab.com/${owner}/${repo}.git`};
  const encodedRepoId = encodeURIComponent(`${owner}/${repo}`);
  const encodedGitTag = encodeURIComponent(nextRelease.gitTag);
  const link = {
    path: 'file.css',
    label: 'README.md',
    type: 'other',
    url: 'https://gitlab.com/gitlab-org/gitlab/-/blob/master/README.md',
  };
  const assets = [link];
  const gitlab = authenticate(env)
    .post(`/projects/${encodedRepoId}/releases`, {
      tag_name: nextRelease.gitTag,
      description: nextRelease.notes,
      assets: {
        links: [
          {
            name: 'README.md',
            url: `https://gitlab.com/${owner}/${repo}/https://gitlab.com/gitlab-org/gitlab/-/blob/master/README.md`,
            link_type: 'other',
          },
        ],
      },
    })
    .reply(200);

  const result = await publish({assets}, {env, cwd, options, nextRelease, logger: t.context.logger});

  t.is(result.url, `https://gitlab.com/${owner}/${repo}/-/releases/${encodedGitTag}`);
  t.deepEqual(t.context.log.args[0], ['Published GitLab release: %s', nextRelease.gitTag]);
  t.true(gitlab.isDone());
});
```

- I had to provide a non empty **path** to a file, otherwise [getAssets](https://github.com/awcjack/gitlab/blob/06915a124346ae8ed0239ffd8f9d18369e579033/lib/publish.js#L33) fails

```bash
  publish › Publish a release with an asset link

  test\publish.test.js:161

   160:
   161:   const result = await publish({assets}, {env, cwd, options, nextRelease, logger: t.context.logger});
   162:

  Rejected promise returned by test. Reason:

  TypeError {
    message: 'Cannot read properties of undefined (reading \'0\')',
  }

  › getPath (node_modules/dir-glob/index.js:8:22)
  › node_modules/dir-glob/index.js:55:50
  › Array.map (<anonymous>)
  › module.exports (node_modules/dir-glob/index.js:54:51)
  › null.<anonymous> (lib/glob-assets.js:17:36)
  › Array.map (<anonymous>)
  › null.module.exports (lib/glob-assets.js:13:18)
  › null.module.exports (lib/publish.js:34:33)
  › test/publish.test.js:161:24

  ─
```

Additionally the created URL will be joined with the URI to the repo the release was created in and the provided URL will be appended, resulting in something like this: `https://gitlab.com/test_user/test_repo/https://gitlab.com/gitlab-org/gitlab/-/blob/master/README.md`

Reply:
Maybe we can filter assets with `url` before we call `getAssets` and handle that beforehand.